qa-tests-backend/src/test/groovy/ClustersTest.groovy

```diff
@@ -16,21 +16,35 @@ class ClustersTest extends BaseSpecification {
 
     def "Test cluster status has cert expiry"() {
         when:
-        "Get the cluster, and the sensor-tls cert"
+        "Get the cluster, and the sensor TLS certificates (sensor-tls and tls-cert-sensor)"
         def cluster = ClusterService.getCluster()
         assert cluster
-        def sensorTLSSecret = orchestrator.getSecret("sensor-tls", "stackrox")
-
-        then:
-        "Verify the cluster has sensor cert expiry information, and that is matches what's in the secret"
         def expiryFromCluster = new Date(
             cluster.getStatus().getCertExpiryStatus().getSensorCertExpiry().getSeconds() * 1000
         )
         assert expiryFromCluster
+
+        def sensorTLSSecret = orchestrator.getSecret("sensor-tls", "stackrox")
         def sensorCert = Cert.loadBase64EncodedCert(sensorTLSSecret.data["sensor-cert.pem"])
         def expiryFromCert = sensorCert.notAfter
         assert expiryFromCert
-        assert expiryFromCert == expiryFromCluster
+
+        def expiryFromNewCert = null
+        try {
+            // tls-cert-sensor is a new secret that may exist in the cluster due to certificate rotation.
+            def newSensorTLSSecret = orchestrator.getSecret("tls-cert-sensor", "stackrox")
+            def newSensorCert = Cert.loadBase64EncodedCert(newSensorTLSSecret.data["cert.pem"])
+            expiryFromNewCert = newSensorCert.notAfter
```

Comment:
This code appears twice how about extracting it to a function?

```groovy
    private def getCertExpiryFromSecret(String secretName, String certKey) {
        def secret = orchestrator.getSecret(secretName, "stackrox")
        def cert = Cert.loadBase64EncodedCert(secret.data[certKey])
        return cert.notAfter
    }
```

Reply:
Done